reduce_boolean triggers in cases where it shouldn't

### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Given 
```swift
struct SSS {
    func reduce(initial: Bool,
                closure: (Bool, Bool) -> Bool) -> Bool {
        false
    }
}
_ = SSS().reduce(initial: false) { _, _ in true }
```

the `reduce_boolean` rule is triggered, and it's suggested I use `contains`.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
[Xcode 14.3's `foo is implemented in bar and baz` lines omitted]
Linting Swift files at paths File.swift
Linting 'File.swift' (1/1)
/Users/eric/File.swift:7:11: warning: Reduce Boolean Violation: Use `contains` instead (reduce_boolean)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version `0.50.3`
* Installation method used `brew`
* No configuration file 
* Which Xcode version are you using: `Xcode 14.3 Build version 14E222b`
* Do you have a sample that shows the issue? `see above`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -40,6 +40,11 @@
 * Fix false positives in `indentation_width` rule.  
   [Sven Münnich](https://github.com/svenmuennich)
 
+* Do not trigger `reduce_boolean` on `reduce` methods with a first named
+  argument that is different from `into`.  
+  [SimplyDanny](https://github.com/SimplyDanny)
+  [#4894](https://github.com/realm/SwiftLint/issues/4894)
+
 ## 0.51.0: bzllint
 
 #### Breaking

--- a/Source/SwiftLintFramework/Rules/Performance/ReduceBooleanRule.swift
+++ b/Source/SwiftLintFramework/Rules/Performance/ReduceBooleanRule.swift
@@ -12,7 +12,8 @@ struct ReduceBooleanRule: SwiftSyntaxRule, ConfigurationProviderRule {
         kind: .performance,
         nonTriggeringExamples: [
             Example("nums.reduce(0) { $0.0 + $0.1 }"),
-            Example("nums.reduce(0.0) { $0.0 + $0.1 }")
+            Example("nums.reduce(0.0) { $0.0 + $0.1 }"),
+            Example("nums.reduce(initial: true) { $0.0 && $0.1 == 3 }")
         ],
         triggeringExamples: [
             Example("let allNines = nums.↓reduce(true) { $0.0 && $0.1 == 9 }"),
@@ -22,7 +23,8 @@ struct ReduceBooleanRule: SwiftSyntaxRule, ConfigurationProviderRule {
             Example("let allNines = nums.↓reduce(true, { $0.0 && $0.1 == 9 })"),
             Example("let anyNines = nums.↓reduce(false, { $0.0 || $0.1 == 9 })"),
             Example("let allValid = validators.↓reduce(true, { $0 && $1(input) })"),
-            Example("let anyValid = validators.↓reduce(false, { $0 || $1(input) })")
+            Example("let anyValid = validators.↓reduce(false, { $0 || $1(input) })"),
+            Example("nums.reduce(into: true) { (r: inout Bool, s) in r = r && (s == 3) }")
         ]
     )
 
@@ -38,6 +40,7 @@ private extension ReduceBooleanRule {
                 let calledExpression = node.calledExpression.as(MemberAccessExprSyntax.self),
                 calledExpression.name.text == "reduce",
                 let firstArgument = node.argumentList.first,
+                firstArgument.label?.text ?? "into" == "into",
                 let bool = firstArgument.expression.as(BooleanLiteralExprSyntax.self)
             else {
                 return

